src/package.ts

```diff
@@ -3,31 +3,48 @@ import { maxSatisfying } from 'semver';
 import got from 'got';
 import { NPMPackage } from './types';
 
+type Package = { version: string; dependencies: Record<string, Package> };
+
 /**
  * Attempts to retrieve package data from the npm registry and return it
  */
 export const getPackage: RequestHandler = async function (req, res, next) {
   const { name, version } = req.params;
-
+  const dependencyTree = {};
   try {
     const npmPackage: NPMPackage = await got(
       `https://registry.npmjs.org/${name}`,
     ).json();
 
-    const dependencies: Record<string, string> = {};
-    for (const [ name, range ] of Object.entries(
-      npmPackage.versions[version].dependencies ?? {},
-    )) {
-      const subPackage: NPMPackage = await got(
-        `https://registry.npmjs.org/${name}`,
-      ).json();
-
-      dependencies[name] =
-        maxSatisfying(Object.keys(subPackage.versions), range) ?? range;
+    const dependencies: Record<string, string> =
+      npmPackage.versions[version].dependencies ?? {};
+    for (const [name, range] of Object.entries(dependencies)) {
+      const subDep = await getDependencies(name, range);
```

Comment:
For me it feels we could consider separate error handling for each call to the registry, to get more granular view if any issue arises.